week-1-ts-ecosystem/w01e03-object-oriented-structure/src/product/Product.ts

```diff
@@ -0,0 +1,27 @@
+import {ProductType, updateProductType} from "../types/product";
+import {TypeCart} from "../cart/Cart";
+
+
+export default class Product<T extends TypeCart> {
+  private product:ProductType<T>
```

Comment:
nie rozumiem trochę tej koncepcji `product` w środku `Product`

Być może było to wymuszone ponieważ chciałeś spełnić założenia `interface`u.
Ten `interface` to główna komplikacja tego rozwiązania. Wystarczy nam tylko klasa.